src/lib.rs

```diff
@@ -523,15 +546,10 @@ fn derive_where_internal(attr: TokenStream, item: TokenStream) -> Result<TokenSt
         let mut where_clause = where_clause.map(|where_clause| where_clause.into_token_stream());
 
         // If there are any bounds, insert them into the `where` clause.
-        if !derive_where.bounds.is_empty() {
+        if let DeriveWhere::WithBounds(bounds, _) = &derive_where {
             // If there is no `where` clause, make one.
             let where_clause = where_clause.get_or_insert(quote! { where });
-
-            for bound in &derive_where.bounds {
-                where_clause.extend(quote! {
-                    #bound: #trait_,
-                })
-            }
+            *where_clause = quote! { #where_clause #(#bounds: #trait_),* };
```

Comment:
I have a `TODO` up there saying that we shouldn't use `quote` here but actually build a proper `WhereClause`, I think we can handle it in this PR.

Reply:
I'll have a look at that